syntax = "proto3";

package NAlice.NScenarios.NGeneralConversation;

option go_package = "a.yandex-team.ru/alice/megamind/protos/analytics/scenarios/general_conversation";
option java_package = "ru.yandex.alice.megamind.protos.analytics.scenarios.general_conversation";

import "mapreduce/yt/interface/protos/extension.proto";

message TProactivityInfo {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    string FrameName = 1 [json_name = "frame_name", (NYT.column_name) = "frame_name"];
    string ActionName = 2 [json_name = "action_name", (NYT.column_name) = "action_name"];
    reserved 3; // MovieToDiscussId
    string EntityKey = 4 [json_name = "entity_key", (NYT.column_name) = "entity_key"];
}

message TDiscussionInfo {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    string EntityKey = 1 [json_name = "entity_key", (NYT.column_name) = "entity_key"];
    bool GivesNegativeFeedback = 2 [json_name = "gives_negative_feedback", (NYT.column_name) = "gives_negative_feedback"];
}

message TFactsCrossPromoInfo {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    string EntityKey = 1 [json_name = "entity_key", (NYT.column_name) = "entity_key"];
}

message TMovieAkinatorResponseInfo {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    reserved 56788;

    message TPosterCloudsState {
        option (NYT.default_field_flags) = SERIALIZATION_YT;

        uint64 LeftNodeId = 1 [(NYT.column_name) = "left_node_id"];
        repeated uint64 ShownLeftMovieIds = 2 [(NYT.column_name) = "shown_left_movie_ids"];
        uint64 RightNodeId = 3 [(NYT.column_name) = "right_node_id"];
        repeated uint64 ShownRightMovieIds = 4 [(NYT.column_name) = "shown_right_movie_ids"];
    }

    message TPosterGalleryState {
        option (NYT.default_field_flags) = SERIALIZATION_YT;

        repeated uint64 ShownMovieIds = 1 [(NYT.column_name) = "shown_movie_ids"];
    }

    message TShowDescriptionState {
        option (NYT.default_field_flags) = SERIALIZATION_YT;

        uint64 MovieId = 1 [(NYT.column_name) = "movie_id"];
    }

    uint64 NodeId = 1 [json_name = "node_id", (NYT.column_name) = "node_id"];
    string ContentType = 2 [json_name = "content_type", (NYT.column_name) = "content_type"];
    string Genre = 3 [json_name = "genre", (NYT.column_name) = "genre"];
    uint64 ShowSimilarMovieId = 4 [json_name = "show_similar_movie_id", (NYT.column_name) = "show_similar_movie_id"];

    oneof State {
        TPosterCloudsState PosterCloudState = 5 [json_name = "poster_cloud_state", (NYT.column_name) = "poster_cloud_state"];
        TPosterGalleryState PosterGalleryState = 6 [json_name = "poster_gallery_state", (NYT.column_name) = "poster_gallery_state"];
        TShowDescriptionState ShowDescriptionState = 7 [json_name = "show_description_state", (NYT.column_name) = "show_description_state"];
    }
}

message TGCResponseInfo {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    string Source = 1 [json_name = "source", (NYT.column_name) = "source"];
    string GcIntent = 2 [json_name = "gc_intent", (NYT.column_name) = "gc_intent", deprecated = true];
    bool IsPureGc = 3 [json_name = "is_pure_gc", (NYT.column_name) = "is_pure_gc"];
    string OriginalIntent = 4 [json_name = "original_intent", (NYT.column_name) = "original_intent", deprecated = true];
    TProactivityInfo ProactivityInfo = 5 [json_name = "proactivity_info", (NYT.column_name) = "proactivity_info"];
    string Intent = 6 [json_name = "intent", (NYT.column_name) = "intent", deprecated = true];
    TDiscussionInfo DiscussionInfo = 7 [json_name = "discussion_info", (NYT.column_name) = "discussion_info"];
    bool HasGifCard = 8 [json_name = "has_gif_card", (NYT.column_name) = "has_gif_card"];
    TFactsCrossPromoInfo FactsCrossPromoInfo = 9 [json_name = "facts_crosspromo_info", (NYT.column_name) = "facts_crosspromo_info"];
    string RecognizedFrame = 10 [json_name = "recognized_frame", (NYT.column_name) = "recognized_frame"];
    string IntentName = 11 [json_name = "intent_name", (NYT.column_name) = "intent_name"];
    uint32 EntitySearchCacheSize = 12 [json_name = "entity_search_cache_size", (NYT.column_name) = "entity_search_cache_size"];
    TMovieAkinatorResponseInfo MovieAkinatorInfo = 13 [json_name = "movie_akinator_info", (NYT.column_name) = "movie_akinator_info"];
    bool IsAggregatedRequest = 14 [json_name = "is_aggregated_request", (NYT.column_name) = "is_aggregated_request"];
    float GcClassifierScore = 15 [json_name = "gc_classifier_score", (NYT.column_name) = "gc_classifier_score"];
}

